x-pack/plugins/lens/public/plugin.ts

```diff
@@ -283,6 +289,15 @@ export class LensPlugin {
 
     visualizations.registerAlias(getLensAliasConfig());
 
+    if (discover) {
+      uiActionsEnhanced.registerDrilldown(
+        new OpenInDiscoverDrilldown({
+          discover,
+          hasDiscoverAccess: () => this.hasDiscoverAccess,
```

Comment:
Might be nice if the dependency on `this.hasDiscoverAccess` was clarified by making it an argument to the `getStartServices` function.

Reply:
Not sure which structure you have in mind - If I'm using `getStartServices` to pull in `hasDiscoverAccess`, then this getter will become async which I'm trying to avoid.